index.d.ts

```diff
@@ -42,17 +42,55 @@ declare class Agent implements Taggable, StartSpanFn {
   currentTraceparent: string | null;
 
   // Transactions
-  startTransaction (
+  startTransaction(
     name?: string,
-    type?: string,
+    options?: TransactionOptions
+  ): Transaction | null;
+  startTransaction(
+    name: string,
+    type: string,
+    options?: TransactionOptions
+  ): Transaction | null;
+  startTransaction(
+    name: string,
+    type: string,
+    subtype: string,
+    options?: TransactionOptions
+  ): Transaction | null;
+  startTransaction(
+    name: string,
+    type: string,
+    subtype: string,
+    action: string,
     options?: TransactionOptions
   ): Transaction | null;
   setTransactionName (name: string): void;
   endTransaction (result?: string | number, endTime?: number): void;
   currentTransaction: Transaction | null;
 
   // Spans
-  startSpan (name?: string, type?: string, options?: SpanOptions): Span | null;
+  startSpan(
+    name?: string,
+    options?: SpanOptions
+  ): Span | null;
+  startSpan(
+    name: string,
```

Comment:
Same as above, if we want to support this:

```js
startSpan(null, 'my-type')
```

Then we should change the definition to:

```suggestion
    name: string | null,
```

Then it does allow `null` values. But this is something that a test should confirm if you add a test with `name` being `null`.

(same goes for the other versions of `startSpan` below...)

Reply:
fixed